src/magicbox/SuggestionsManager.ts

```diff
@@ -306,29 +336,85 @@ export class SuggestionsManager {
     return $$(dom);
   }
 
-  private move(direction: 'up' | 'down') {
+  private move(direction: direction) {
+    const previewSelectables = $$(this.element).findAll(`.coveo-preview-selectable`);
+    if (previewSelectables.length > 0) {
+      return this.moveWithQuerySuggestPreview(direction);
+    }
+    return this.moveWithinSuggestion(direction);
+  }
+
+  private moveWithinSuggestion(direction: direction) {
     const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
     const selectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
     const currentIndex = indexOf(selectables, currentlySelected);
 
     let index = direction == 'up' ? currentIndex - 1 : currentIndex + 1;
-    if (index < -1) {
-      index = selectables.length - 1;
-    }
-    if (index > selectables.length) {
-      index = 0;
-    }
+    index = (index + selectables.length) % selectables.length;
```

Comment:
Why do we need to add `selectables.length` before calculating the modulus? Would this be equivalent?
```
index = index % selectables.length
```

I would also prefer storing the result in a new variable rather than mutating the `index` variable which could become a `const`.

Reply:
No it will not be the same. In JS, the modulus is not working with negative value. you can try it in your console.

![Capture d’écran, le 2019-08-15 à 17 33 41](https://user-images.githubusercontent.com/50176606/63169510-fb4c6200-c004-11e9-9d91-4c88fa767b71.png)

I will store it in another variable

Reply:
Nice workaround you did there!